### PR TITLE
512/responsive-enhancement-events

### PR DESCRIPTION
**Pull request by sxpydo**
Ticket: https://github.com/isaaccomputerscience/isaac-cs-issues/issues/512
Figma design: https://www.figma.com/design/xLvYVdW2miyMeZk2QuJ0uU/Isaac---Designs---240412

### CODE REVIEW

**Review by KrisBoxAND — CHANGES_REQUESTED** [review]
just one from me here :)

File: src/app/components/elements/Carousel.tsx [hunk 1]
```diff
@@ -88,28 +89,44 @@ const ControlledCarouselInstance = ({ children, collectionTag }: any) => {
...
-    groupingLimit = 3;
+  let effectiveGroupingLimit = groupingLimit || GROUP_LIMITS.DEFAULT;
+
+  if (screenWidth >= SCREEN_SIZES.MOBILE && screenWidth <= SCREEN_SIZES.TABLET) {
+    effectiveGroupingLimit = GROUP_LIMITS.TABLET;
+  } else if (screenWidth < SCREEN_SIZES.MOBILE) {
+    effectiveGroupingLimit = GROUP_LIMITS.MOBILE;
   }
 
+  const tuple: any = [];
```
[the remark this answers is not in the document]
Comment (sxpydo) [reply src/app/components/elements/Carousel.tsx:108]: @KrisBoxAND I didn't write that line of the code, I only moved it to where `tuple` is used for clarity, I am happy to move it back to line 91.

File: src/app/components/elements/Carousel.tsx [hunk 2]
```diff
@@ -88,28 +89,49 @@ const ControlledCarouselInstance = ({ children, collectionTag }: any) => {
...
   }
 
-  children.forEach((child: any, index: number) => {
-    if (index % groupingLimit === 0) {
-      tuple.push([]);
-    }
-    tuple[Math.floor(index / groupingLimit)].push(child);
-  });
+  type TupleItem = React.ReactNode[];
+  const [tuple, setTuple] = useState<TupleItem[]>([]);
```
Comment (KrisBoxAND) [inline src/app/components/elements/Carousel.tsx:109]: is this type right? Just to double check it's a nested array?

**Review by KrisBoxAND — COMMENTED** [review]
one last question, will approve if proven wrong :D

File: src/app/components/elements/Carousel.tsx (hunk 2 above)
Comment (sxpydo) [reply src/app/components/elements/Carousel.tsx:109]: You're not wrong, It's a nested array 👀 I'm going to guess that there is an even better way to improve this 😅

Comment (KrisBoxAND) [reply src/app/components/elements/Carousel.tsx:109]: Actually, it's good - I just wanted to double check :)

**Review by KrisBoxAND — APPROVED** [review]

**Pull request merged**